src/Nethermind/Nethermind.Network.Discovery/Discv5/NettyDiscoveryV5Handler.cs

```diff
@@ -0,0 +1,67 @@
+// SPDX-FileCopyrightText: 2024 Demerzel Solutions Limited
+// SPDX-License-Identifier: LGPL-3.0-only
+
+using System.Net;
+using System.Net.Sockets;
+using System.Threading.Channels;
+using DotNetty.Buffers;
+using DotNetty.Transport.Channels;
+using DotNetty.Transport.Channels.Sockets;
+using Lantern.Discv5.WireProtocol.Connection;
+using Microsoft.Extensions.DependencyInjection;
+using Nethermind.Logging;
+using Nethermind.Serialization.Rlp;
+
+namespace Nethermind.Network.Discovery;
+
+/// <summary>
+/// Adapter, integrating DotNetty externally-managed <see cref="IChannel"/> with Lantern.Discv5
+/// </summary>
+public class NettyDiscoveryV5Handler : NettyDiscoveryBaseHandler, IUdpConnection
+{
+    private readonly ILogger _logger;
+    private readonly Channel<UdpReceiveResult> _inboundQueue;
+
+    private IChannel? _nettyChannel;
+
+    public NettyDiscoveryV5Handler(ILogManager loggerManager) : base(loggerManager)
+    {
+        _logger = loggerManager.GetClassLogger<NettyDiscoveryV5Handler>();
+        _inboundQueue = Channel.CreateUnbounded<UdpReceiveResult>();
```

Comment:
NP: Might cause memory leak in case reader is too slow or hang or crash. Probably set a somewhat high bound. The writer already use TryWrite, so thats fine.